### DISCUSSION TITLE
Make plugin creation more structured to make discovery easier

### DISCUSSION THREAD

**Discussion by chrisdmacrae** [11ty/eleventy#4157] [Enhancement Queue]
A lot of frameworks are now namespacing their plugins as a requirement, to make discovery easier and installation more magical.

E.g, gatsby requires:

- `gatsby-source-*`
- `gatsby-transformer-*`
- `gatsby-plugin-*`

etc

Parcel requires `parcel-plugin-*`.

I propose eleventy does the same thing, with `eleventy-plugin-*` or `11ty-plugin-*`.

**Comment by chrisdmacrae** [comment]
@zachleat I'm going to draft an update to the plugins page detailing how to develop a bunch of different plugins. 

I'll also pull in any found plugins on NPM and list them.

**Comment by zachleat** [comment]
Beautiful idea. I’d stick with exclusively using `eleventy-plugin-*` though if we can.

**Comment by Ryuno-Ki** [comment]
You could even ~steal~ copy ChaiJS' approach and list plugins automatically: https://github.com/chaijs/chaijs.github.io/issues/173#issuecomment-389127171

I looked around some more about how that scraping could be done and discovered [npmsearch](https://github.com/nodesource/npmsearch#api).

A query could look like https://npmsearch.com/query?q=eleventy-plugin&fields=keywords,name (for some reason, neither `homepage` nor `repository` returns a meaningful value.

I wouldn't expect problems if you don't jam their service … asking them is a nice gesture, though :-)

**Comment by Ryuno-Ki** [comment]
@zachleat Could you setup a 11ty/awesome-eleventy GitHub repo where we could PR plugins in?

**Comment by chrisdmacrae** [comment]
Heads up -- this is in the works, will try and push up a viable MVP asap.

**Comment by Ryuno-Ki** [comment]
@chrisdmacrae As stop-gap I went through npm and listed packages I found there.

**Comment by xplosionmind** [comment]
Hey all, I am resuscitating this issue after five years (😲) because I would really love to create some simple plugins. Unfortunately, my knowledge of JavaScript is very minimal, and I am not able to reverse-engineer Eleventy’s APIs and structure to understand where to start, what are the arguments passed, what is returned and where, etc…

In short, I lack all the basics, and I believe it would be super valuable if there could be an explanatory article (or video) that could go through the essentials to understand how Eleventy works behind the scenes.

Any links or suggestions for Eleventy plugin creation 101?

**Reply by zachleat** [reply]
Sorry for the late reply but this exists here: https://www.11ty.dev/docs/create-plugin/
